### ISSUE TITLE
Saving the uploaded file to a permanent location, not working on mobile devices.

### ISSUE THREAD

**Issue by Pieter3033** [lovasoa/SQLpage#199]
Hi all,

`sqlpage.uploaded_file_path` Returning `NULL` when submitting on a mobile device.

**On computer:**
Works perfectly fine, file path is inserted into database.

**Code:**
```
INSERT INTO Images (JC_num, file_path, title)
VALUES(
        $
       ,sqlpage.uploaded_file_path('img')
       ,'test'
)
```

**Results:** (Reference from DB)
![image](https://github.com/lovasoa/SQLpage/assets/141907966/559d19ed-afa7-48dc-bd17-80ca85a33367)

**On Mobile device:**

Using the exact same code to save the image locally.
The file path just returns as NULL.

**Reference from DB:**
![image](https://github.com/lovasoa/SQLpage/assets/141907966/6c55ff5d-c62c-4316-982d-b0310583a637)

### To Reproduce
1. Create a webpage with a form to submit images.
2. Host the website so that it can be accessed from a mobile device.
3. Using the mobile device upload the `file_path` to the database.

The file_path will return as NULL.

**Comment by lovasoa** [comment]
Hi! It looks like the mobile device did not send the file. Did you try from a different browser or device ?

On a separate note: it looks like you are saving temporary file paths in a permanent store (the database). Are you sure this is really what you want to do? You probably want to save files in a persistent location, where they won't be automatically deleted, and save that permanent path in the database.

**Comment by Pieter3033** [comment]
> On a separate note: it looks like you are saving temporary file paths in a permanent store (the database). Are you sure this is really what you want to do? You probably want to save files in a persistent location, where they won't be automatically deleted, and save that permanent path in the database.

I have moved my file first and then inserted the path into the database. It has moved the file to a persistent location, but it inserted the temp file location in the DB, am I missing something?

**Code:**
```
SET image1_path = sqlpage.uploaded_file_path('reg_img')
SET exec_image = sqlpage.exec( 'cmd','/C','move', $image1_path, 'C:\SQLPage\Jobcard_Images');

INSERT INTO Images (JC_num, file_path, title)
VALUES(
        $
       ,sqlpage.uploaded_file_path('reg_img')
       ,'reg_no'
)
```

**Result:**
![image](https://github.com/lovasoa/SQLpage/assets/141907966/ef8d12e2-6fb7-4f6e-b596-298bbef6a618)

**Comment by lovasoa** [comment]
You are saving the `uploaded_file_path` instead of the new path you created. 

What you can do is 

```sql
SET image1_path = sqlpage.uploaded_file_path('reg_img');
SET target_file_path = 'C:\SQLPage\Jobcard_Images/' || sqlpage.random_string(16) || '.jpg';
SET exec_image = sqlpage.exec('cmd','/C','move', $image1_path, $target_file_path);

INSERT INTO Images (JC_num, file_path, title)
VALUES($, $target_file_path, 'reg_no');
```

**Comment by lovasoa** [comment]
I haven't been able to reproduce your issue. Maybe you can share an small sample application that reliably reproduces the issue ?

Or [a network capture from the mobile device](https://developer.chrome.com/docs/devtools/remote-debugging) showing the issue ?

**Comment by lovasoa** [comment]
v0.20.1 adds a function that should make your life easier: https://sql.ophir.dev/functions.sql?function=persist_uploaded_file#function

and it doesn't silently ignore large files

**Comment by Pieter3033** [comment]
Hi @lovasoa, trust you are well.

With the new update I know exactly what the problem is, it was not an error with the browser nor the device being used.

I did not know at the time but the `max_uploaded_file_size` is the max size per webpage, and not per image.

You are correct, I did not test with the same files every time.

The solution I found is to download an app that automatically lessens the image size.
With this solution I am able to upload multiple images on a single page.

SQLPage `persist_uploaded_file` definitely makes things easier.
